Modelica/Blocks/Math.mo

```diff
@@ -2264,37 +2264,64 @@ explicitly set to 0.0, if the mean value results in a negative value.
 
-  block RectifiedMean "Calculate rectified mean over period 1/f"
+  block TriggeredMean "Calculate mean over period determined by trigger"
     extends Modelica.Blocks.Interfaces.SISO;
-    parameter SI.Frequency f(start=50) "Base frequency";
+    import Modelica.Constants.eps;
     parameter Real x0=0 "Start value of integrator state";
-    parameter Real y0(final min=0)=0 "Start value of output";
-    Mean mean(final f=f, final x0=x0,
-      final y0=y0)
-      annotation (Placement(transformation(extent={{0,-10},{20,10}})));
-    Blocks.Math.Abs abs1
-      annotation (Placement(transformation(extent={{-60,-10},{-40,10}})));
+    parameter Real y0=0 "Start value of output";
+    parameter Boolean yGreaterOrEqualZero=false
+      "= true, if output y is guaranteed to be >= 0 for the exact solution"
+      annotation (Evaluate=true, Dialog(tab="Advanced"));
+    Modelica.Blocks.Interfaces.BooleanInput trigger "Trigger input signal"
+      annotation (Placement(
+          transformation(
+          extent={{-20,-20},{20,20}},
+          rotation=90,
+          origin={0,-120})));
+  protected
+    Modelica.Units.SI.Time t0 "Last trigger instance";
+    Real x "Integrator state";
+    discrete Real y_last "Last sampled mean value";
+  initial equation
+    pre(trigger)=false;
+    t0 = time;
+    x = x0;
+    y_last = y0;
   equation
-    connect(u, abs1.u) annotation (Line(
-        points={{-120,0},{-62,0}}, color={0,0,127}));
-    connect(abs1.y, mean.u) annotation (Line(
-        points={{-39,0},{-2,0}}, color={0,0,127}));
-    connect(mean.y, y) annotation (Line(
-        points={{21,0},{110,0}}, color={0,0,127}));
+    der(x) = u;
+    when edge(trigger) then
+      y_last = if noEvent((time - pre(t0))>eps) then
+        if not yGreaterOrEqualZero then pre(x)/(time - pre(t0))
+          else max(0.0, pre(x)/(time - pre(t0)))
+        else pre(y_last);
+      t0 = time;
+      reinit(x, 0);
+    end when;
+    y = y_last;
     annotation (Documentation(info="<html>
 <p>
-This block calculates the rectified mean of the input signal u over the given period 1/f, using the
-<a href=\"modelica://Modelica.Blocks.Math.Mean\">mean block</a>.
+This block calculates the mean of the input signal u over the period dtermined by the edges of the trigger:
 </p>
+<blockquote><pre>
+ 1   T
+---- &int; u(t) dt
+T-t0 t0
+</pre></blockquote>
 <p>
-Note: The output is updated after each period defined by 1/f.
+Note: The output is updated at each trigger instance.
+</p>
+
+<p>
+If parameter <strong>yGreaterOrEqualZero</strong> in the Advanced tab is <strong>true</strong> (default = <strong>false</strong>),
+then the modeller provides the information that the mean of the input signal is guaranteed
+to be &ge; 0 for the exact solution. However, due to inaccuracies in the numerical integration scheme,
+the output might be slightly negative. If this parameter is set to true, then the output is
+explicitly set to 0.0, if the mean value results in a negative value.
```

Comment:
I am afraid I do not understand the reason for the implementation of `yGreaterOrEqualZero`:
- Is it a side effect that has to be suppressed or is this just a "normal" integration phenomenon?
- It possibly shall be explained under which circumstances a slightly negative output occurs
- What happens if -- in the above case -- the negative input is fed to the block in a simulation model: 
    - Is then the output slightly positive (instead of negative)? 
    - Shall we also implement countermeasure for this case?  

### Simulation results

- Possibly some analytical reference signals shall be provided and compared to the triggered block calculations.
- We shall then also explain in the documentation what signals shall be compared to each other.

Reply:
This was Introduced some time ago (I'm not sure who did that). The reason is simple:
The result of mean could be slightly negative due to numerical reasons, although the analytical result is zero. 
This is annoying since Mean is used in RootMeanSquare, and RMS can only be greater or euqal to zero.

Reply:
> ### Simulation results
> * Possibly some analytical reference signals shall be provided and compared to the triggered block calculations.
> * We shall then also explain in the documentation what signals shall be compared to each other.

@AHaumer this shall be provided to the users to better understand the purpose of the example(s)